tools/alias.js

```diff
@@ -63,19 +63,23 @@ function updateAliases (aliases, indices) {
   return client.indices.updateAliases({ body: { actions } })
 }
 
-function run (cluster, command) {
+async function run (cluster, command) {
   client = elastic(cluster)
 
-  return Promise.all([ fetchAliases(), fetchIndices() ])
-    .then(([ aliases, indices ]) => updateAliases(aliases, indices))
-    .then(() => {
-      console.log('Alias complete')
-      process.exit()
-    })
-    .catch((err) => {
-      console.error(`Alias failed: ${err.toString()}`)
-      process.exit(1)
-    })
+  const [ aliases, indices ] = await Promise.all([
+    fetchAliases(),
+    fetchIndices()
+  ])
+
+  try {
+    await updateAliases(aliases, indices)
+
+    console.log('Alias complete')
+    process.exit()
+  } catch (err) {
+    console.error(`Alias failed: ${err.toString()}`)
```

Comment:
I'm curious, with async/await, instead of putting them in a try catch could you do something like
```js
async function run(cluster, command) {
    client = elastic(cluster)
    
    const [aliases, indices] = await Promise.all([
        fetchAliases(),
        fetchIndices()
    ])
    
    await updateAliases(aliases, indices)
        .catch(err => {
            console.error(`Alias failed: ${err.toString()}`);
            return process.exit(1);
        })

    console.log('Alias complete');
    process.exit();

}

```

Reply:
Yeah, that's legit. I quite like mixing up `.then()` and `.catch()` with `await`.

TBH my refactor here isn't great, any errors when fetching the aliases/indices will be lost into the abyss... the original promise chain may actually be more readable.

Reply:
Of course I could be wrong and all my async/await code sucks... who knows.